modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/results/portlet/shared/search/SearchResultsPortletSharedSearchContributor.java

```diff
@@ -81,34 +79,29 @@ private void _paginate(
 		searchRequestBuilder.paginationStartParameterName(
 			paginationStartParameterName);
 
-		Optional<String> paginationStartParameterValueOptional =
-			portletSharedSearchSettings.getParameterOptional(
-				paginationStartParameterName);
+		String paginationDeltaString = portletSharedSearchSettings.getParameter(
+			searchResultsPortletPreferences.getPaginationDeltaParameterName());
 
-		SearchOptionalUtil.copy(
-			() -> paginationStartParameterValueOptional.map(Integer::valueOf),
-			portletSharedSearchSettings::setPaginationStart);
+		Integer paginationDelta = (paginationDeltaString != null) ?
+			Integer.valueOf(paginationDeltaString) :
+				searchResultsPortletPreferences.getPaginationDelta();
```

Comment:
@dantewang I'm using ternary here to avoid declaring Integer paginationDelta to null first